web/react/components/sidebar.jsx

```diff
@@ -235,103 +247,143 @@ module.exports = React.createClass({
             }
         }
     },
+    onScroll: function(e) {
+        this.updateUnreadIndicators();
+    },
+    onResize: function(e) {
+        this.updateUnreadIndicators();
+    },
+    updateUnreadIndicators: function() {
+        var container = $(this.refs.container.getDOMNode());
+
+        if (this.firstUnreadChannel) {
```

Comment:
No problem at all. I actually had to test it myself to make sure it worked as I thought it did.